actors/common/src/main/java/org/sunbird/learner/actors/bulkupload/OrgBulkUploadBackGroundJobActor.java

```diff
@@ -0,0 +1,180 @@
+import org.sunbird.actor.router.ActorConfig;
+import org.sunbird.actorutil.org.OrganisationClient;
+import org.sunbird.actorutil.org.impl.OrganisationClientImpl;
+import org.sunbird.common.Constants;
+import org.sunbird.common.exception.ProjectCommonException;
+import org.sunbird.common.models.util.*;
+import org.sunbird.common.request.ExecutionContext;
+import org.sunbird.common.request.Request;
+import org.sunbird.common.responsecode.ResponseCode;
+import org.sunbird.learner.actors.bulkupload.model.BulkUploadProcess;
+import org.sunbird.learner.actors.bulkupload.model.BulkUploadProcessTask;
+import org.sunbird.learner.util.Util;
+
+/** Created by rajatgupta on 15/10/18. */
+@ActorConfig(
+  tasks = {"orgBulkUploadBackground"},
+  asyncTasks = {}
+)
+public class OrgBulkUploadBackGroundJobActor extends BaseBulkUploadBackGroundJobActor {
+  private OrganisationClient orgClient = new OrganisationClientImpl();
+
+  @Override
+  public void onReceive(Request request) throws Throwable {
+    String operation = request.getOperation();
+    Util.initializeContext(request, TelemetryEnvKey.ORGANISATION);
+    ExecutionContext.setRequestId(request.getRequestId());
+    if (operation.equalsIgnoreCase("orgBulkUploadBackground")) {
+      handleBulkUploadBackround(
+          request,
+          (baseBulkUpload) -> {
+            processBulkUpload(
+                (BulkUploadProcess) baseBulkUpload,
+                (task) -> {
+                  processOrg((BulkUploadProcessTask) task);
+                  return null;
+                });
+            return null;
+          });
+    } else {
+      onReceiveUnsupportedOperation("OrgBulkUploadBackGroundJobActor");
+    }
+  }
+
+  @Override
+  public void processBulkUpload(BulkUploadProcess bulkUploadProcess, Function function) {
+    Integer sequence = 0;
+    Integer taskCount = bulkUploadProcess.getTaskCount();
+    List<Map<String, Object>> successList = new LinkedList<>();
+    List<Map<String, Object>> failureList = new LinkedList<>();
+    List<BulkUploadProcessTask> nonRootOrg = new ArrayList<>();
+    while (sequence <= taskCount) {
+      Integer nextSequence = sequence + CASSANDRA_BATCH_SIZE;
+      Map<String, Object> queryMap = new HashMap<>();
+      queryMap.put(JsonKey.PROCESS_ID, bulkUploadProcess.getId());
+      Map<String, Object> sequenceRange = new HashMap<>();
+      sequenceRange.put(Constants.GT, sequence);
+      sequenceRange.put(Constants.LTE, nextSequence);
+      queryMap.put(BulkUploadJsonKey.SEQUENCE_ID, sequenceRange);
+      List<BulkUploadProcessTask> tasks = bulkUploadProcessTaskDao.readByPrimaryKeys(queryMap);
+      for (BulkUploadProcessTask task : tasks) {
+        try {
+          if (task.getStatus() != null
+              && task.getStatus() != ProjectUtil.BulkProcessStatus.COMPLETED.getValue()) {
+            Map<String, Object> row = mapper.readValue(task.getData(), Map.class);
+            if (row.containsKey(JsonKey.IS_ROOT_ORG)
```

Comment:
Restructure the code so that type (e.g. org) specific code is moved to type specific handler (e.g. processOrg) and the base class method is called.